easyrelease-lib/src/main/groovy/eu/inloop/easyrelease/plugin/Util.groovy

```diff
@@ -0,0 +1,142 @@
+
+            def engine = new SimpleTemplateEngine()
+            def fileName = engine.createTemplate(template).make(binding).toString() + ".apk"
+
+            try {
+                variant.outputs.each { output ->
+                    output.outputFile = new File(output.outputFile.parent, fileName)
+                    println "$TAG Setting $output.name variant output name to $fileName"
+                }
+            } catch (all) {
+                println "$TAG Exception raised during build: " + all
+            }
+        }
+    }
+
+    static def getNameString(project) {
+        def projectDir = project.projectDir
+        def propFileName = 'easyrelease.properties'
+        def propFile = new File("$projectDir/$propFileName")
+        if (propFile.canRead()) {
+            def props = new Properties()
+            props.load(new FileInputStream(propFile))
+
+            if (props["APK_NAME"] != null) {
+                return props["APK_NAME"]
+            }
+            println "$TAG APK_NAME is null, using the default one "
+        }
+        return DEFAULT_APK_STRUCTURE
+    }
+
+    static def getBinding(project, variant) {
+        def now = Calendar.getInstance()
+        def binding = [
+                projectName : project.name,
+                variantName : variant.name,
+                // first check version code and name from Gradle build script, then from AndroidManifest.xml
+                variantVersionCode : variant.versionCode ? variant.versionCode : getVersionCode(project),
+                variantVersionName : variant.versionName ? variant.versionName : getVersionName(project),
+                dateYear : now.get(Calendar.YEAR),
+                dateMonth : now.get(Calendar.MONTH) + 1,
+                dateDay : now.get(Calendar.DAY_OF_MONTH),
+                dateHour : now.get(Calendar.HOUR_OF_DAY),
+                dateMinute : now.get(Calendar.MINUTE),
+                dateSecond : now.get(Calendar.SECOND),
+                dateMillisecond : now.get(Calendar.MILLISECOND)
+        ]
+
+        try {
+            def repo = Grgit.open(project.file('..'))
+            // buildNumber based on branch commit count might crash when git path is
+            // set incorrectly, leaving out until further research
+            // binding.buildNumber = 'git rev-list --count HEAD'.execute().text.trim()
+            binding.headShortHash = repo.head().abbreviatedId
+            binding.headAuthorName = repo.head().author.getName()
+            binding.headAuthorEmail = repo.head().author.getEmail()
+            binding.headCommiterName = repo.head().committer.getName()
+            binding.headCommiterEmail = repo.head().committer.getEmail()
+            def branchFullName = repo.branch.current.name.split("//")
+            binding.branchName = branchFullName[branchFullName.length-1]
+            binding.each {
+                println (it)
+            }
+            return binding
+        } catch (all) {
```

Comment:
Catching exception and just printing info about them does not seem like a valid exception handling here. If the exception would be thrown just because of not existing .git directory, then ignore it and dont print anything, or print nice info that git wasnt found, or something in this spirit.